pkg/apis/pingcap.com/v1alpha1/types.go

```diff
@@ -260,3 +260,215 @@ type TiKVFailureStore struct {
+	CommitTs   string            `json:"commitTs"`
+	Conditions []BackupCondition `json:"conditions"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupSchedule is a backup schedule of tidb cluster.
+type BackupSchedule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   BackupScheduleSpec   `json:"spec"`
+	Status BackupScheduleStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupScheduleList contains a list of BackupSchedule.
+type BackupScheduleList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []BackupSchedule `json:"items"`
+}
+
+// BackupScheduleSpec contains the backup schedule specification for a tidb cluster.
+type BackupScheduleSpec struct {
+	// Schedule specifies the cron string used for backup scheduling.
+	Schedule string `json:"schedule"`
+	// MaxBackups is to specify how many backups we want to keep.
+	MaxBackups int `json:"maxBackups"`
+	// BackupTemplate is the specification of the backup structure to get scheduled.
+	BackupTemplate BackupSpec `json:"backupTemplate"`
+}
+
+// BackupScheduleStatus represents the current state of a BackupSchedule.
+type BackupScheduleStatus struct {
+	// LastBackup represents the last backup.
+	LastBackup string `json:"lastBackup"`
+	// LastBackupTime represents the last time the backup was successfully created.
+	LastBackupTime metav1.Time `json:"lastBackupTime"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Restore represents the restoration of backup of a tidb cluster.
+type Restore struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   RestoreSpec   `json:"spec"`
+	Status RestoreStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RestoreList contains a list of Restore.
+type RestoreList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []Restore `json:"items"`
+}
+
+// RestoreConditionType represents a valid condition of a Restore.
+type RestoreConditionType string
+
+const (
+	// RestoreRunning means the Restore is currently being executed.
+	RestoreRunning RestoreConditionType = "Running"
+	// RestoreComplete means the Restore has successfully executed and the
+	// backup data has been loaded into tidb cluster.
+	RestoreComplete RestoreConditionType = "Complete"
+	// RestoreFailed means the Restore has failed.
+	RestoreFailed RestoreConditionType = "Failed"
+)
+
+// RestoreCondition describes the observed state of a Restore at a certain point.
+type RestoreCondition struct {
+	Type    RestoreConditionType   `json:"type"`
+	Status  corev1.ConditionStatus `json:"status"`
+	Reason  string                 `json:"reason"`
+	Message string                 `json:"message"`
```

Comment:
this field should add `omitempty` tag

Reply:
It doesn't matter, address it in the next PR